packages/grid/src/GridUtils.ts

```diff
@@ -745,90 +745,107 @@ export class GridUtils {
     visibleIndex: VisibleIndex,
     movedItems: MoveOperation[] = []
   ): ModelIndex {
-    let modelIndex = visibleIndex;
-
-    for (let i = movedItems.length - 1; i >= 0; i -= 1) {
-      const movedItem = movedItems[i];
-      if (modelIndex === movedItem.to) {
-        ({ from: modelIndex } = movedItem);
-      } else if (movedItem.from <= modelIndex && modelIndex < movedItem.to) {
-        modelIndex += 1;
-      } else if (movedItem.to < modelIndex && modelIndex <= movedItem.from) {
-        modelIndex -= 1;
-      }
-    }
+    const modelIndex = GridUtils.applyItemMoves(
+      visibleIndex,
+      visibleIndex,
+      movedItems,
+      true
+    )[0][0];
 
     return modelIndex;
   }
 
   /**
-   * Translate the provided UI start/end indexes to the model start/end indexes by applying the `movedItems` transformations.
-   * Since moved items can split apart a range, multiple pairs of indexes are returned
-   *
-   * @param start Start item in one dimension
-   * @param end End item in one dimension
-   * @param movedItems Moved item pairs in this dimension
-   * @returns Array of start/end pairs of the indexes after transformations applied.
+   * Applies the items moves to the 1D range
+   * @param start The start index of the range
+   * @param end The end index of the range
+   * @param movedItems The move operations to apply
+   * @param reverse If the moved items should be applied in reverse (this reverses the effects of the moves)
+   * @returns A list of 1D ranges in the translated space. Possibly multiple non-continuous ranges
    */
-  static getModelRangeIndexes(
+  private static applyItemMoves(
+    start: number,
+    end: number,
+    movedItems: MoveOperation[],
+    reverse?: boolean
+  ): [number, number][];
+
+  // Without this duplicate signature, TS will complain
+  private static applyItemMoves(
     start: GridRangeIndex,
     end: GridRangeIndex,
-    movedItems: MoveOperation[]
+    movedItems: MoveOperation[],
+    reverse?: boolean
+  ): GridAxisRange[];
```

Comment:
I was trying to think of a way to write this with generics instead, but couldn't figure it out :(

Reply:
I got something to work. Requires a cast at the return, but the rest of it seems to work

The reason I added the overloads was the `getVisibleIndex` and `getModelIndex` method calls `applyItemMoves` w/ a known non-null value. But previously this always returned possibly null values in the array. So I needed a null check or assertion that _should_ never fail